c7n/resources/rds.py

```diff
@@ -1189,11 +1190,85 @@ def process_resource_set(self, resource_set):
             shared_accounts = set(attrs.get('restore', []))
             delta_accounts = shared_accounts.difference(self.accounts)
             if delta_accounts:
-                r['c7n:CrossAccountViolations'] = list(delta_accounts)
+                r[self.annotation_key] = list(delta_accounts)
                 results.append(r)
         return results
 
 
+@RDSSnapshot.action_registry.register('remove-restore-permissions')
```

Comment:
let's yank the restore from the name, there aren't any other permissions on snapshots in general, i think we've generally opted to model the action as `set-permissions` with the ability to add as well (ebs-snapshot.set-permissions for example).

Reply:
Thanks, I wasn't aware of that action. Reworked this PR to be more consistent with the EBS behavior.